manim/animation/animation.py

```diff
@@ -192,29 +299,92 @@ def get_sub_alpha(self, alpha: float, index: int, num_submobjects: int) -> float
 
     # Getters and setters
     def set_run_time(self, run_time: float) -> "Animation":
+        """Set the run time of the animation.
+
+        Parameters
+        ----------
+        run_time
+            The new time the animation should take in seconds.
+
+        .. note::
+
+            The run_time of an animation should not be changed while it is already
+            running.
+
+        Returns
+        -------
+        ``Animation``
+            ``self``
+        """
         self.run_time = run_time
         return self
 
     def get_run_time(self) -> float:
+        """Get the run time of the animation.
+
+        Returns
+        -------
+        float
+            The time the animation takes in seconds.
+        """
         return self.run_time
 
     def set_rate_func(
         self,
         rate_func: Callable[[float], float],
     ) -> "Animation":
+        """Set the rate function of the animation.
+
+        Parameters
+        ----------
+        run_time
+            The new time the animation should take in seconds.
+
+
+        Returns
+        -------
+        ``Animation``
+            ``self``
+        """
         self.rate_func = rate_func
         return self
 
     def get_rate_func(
         self,
     ) -> Callable[[float], float]:
+        """Get the rate function of the animation.
+
+        Returns
+        -------
+        Callable[[float], float]
```

Comment:
```suggestion
```
These lines are not necessary, as this is already defined by the typehint -> Callable[[float], float]:

Reply:
I tested it and it actually does not work without it I'm afraid.

Reply:
okay, I did not know that.
But then, I am also not sure why we have this typehint at all.

Reply:
For Intellisense I guess :man_shrugging:

Reply:
Ah,  makes sense